RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducer.cc

```diff
@@ -8,55 +8,87 @@
 #include "FWCore/Utilities/interface/EDGetToken.h"
 #include "FWCore/Utilities/interface/InputTag.h"
 #include "FWCore/Utilities/interface/StreamID.h"
-#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/SynchronizingEDProducer.h"
 #include "HeterogeneousCore/CUDACore/interface/JobConfigurationGPURecord.h"
 #include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
 #include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusterParamsDeviceCollection.h"
 #include "RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.h"
 #include "RecoParticleFlow/PFRecHitProducer/interface/PFRecHitTopologyRecord.h"
 
 namespace ALPAKA_ACCELERATOR_NAMESPACE {
-  class PFClusterSoAProducer : public stream::EDProducer<> {
+  class PFClusterSoAProducer : public stream::SynchronizingEDProducer<> {
   public:
     PFClusterSoAProducer(edm::ParameterSet const& config)
         : pfClusParamsToken(esConsumes(config.getParameter<edm::ESInputTag>("pfClusterParams"))),
           topologyToken_(esConsumes(config.getParameter<edm::ESInputTag>("topology"))),
           inputPFRecHitSoA_Token_{consumes(config.getParameter<edm::InputTag>("pfRecHits"))},
           outputPFClusterSoA_Token_{produces()},
           outputPFRHFractionSoA_Token_{produces()},
+          num_rhf_{cms::alpakatools::make_host_buffer<uint32_t>()},
```

Comment:
This should be
```suggestion
          num_rhf_{cms::alpakatools::make_host_buffer<uint32_t, Platform>()},
```
to use pinned host memory that can be efficiently accessed by the devices, _e.g._ for `memcpy` or to write directly to.

Reply:
Made this change.